test/transaction.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+    This file is part of cpp-ethereum.
+
+    cpp-ethereum is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    cpp-ethereum is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file transaction.cpp
+ * @author Christoph Jentzsch <cj@ethdev.com>
+ * @date 2014
+ * Transaaction test functions.
+ */
+
+#include <boost/filesystem/operations.hpp>
+#include <boost/test/unit_test.hpp>
+#include "JsonSpiritHeaders.h"
+#include <libdevcore/CommonIO.h>
+#include <libethereum/BlockChain.h>
+#include <libethereum/State.h>
+#include <libethereum/ExtVM.h>
+#include <libethereum/Defaults.h>
+#include <libevm/VM.h>
+#include "TestHelper.h"
+
+using namespace std;
+using namespace json_spirit;
+using namespace dev;
+using namespace dev::eth;
+using namespace dev::eth;
+
+namespace dev {  namespace test {
+
+bytes importByteArray(const std::string& str)
+{
+    bytes data;
+    if (str.find_first_of("0x") == 0)
+            data = fromHex(str.substr(2));
+        else
+            data = fromHex(str);
+    return data;
+}
+
+void doTransactionTests(json_spirit::mValue& v, bool _fillin)
+{
+    processCommandLineOptions();
+
+    for (auto& i: v.get_obj())
+    {
+        cerr << i.first << endl;
+        mObject& o = i.second.get_obj();
+
+        BOOST_REQUIRE(o.count("transaction") > 0);
```

Comment:
if it is an invalid transaction this is not here. You need to differentiate more between _fillin = true (only transaction will be there) and _fillin=false (only rlp is rquired, transaction and sender are only present for valid transactions)